### PR TITLE
Remove useless parts of code in SemanticModel class

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/SimpleClassNameCheck.java [hunk 1]
```diff
@@ -33,6 +31,9 @@
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.Tree.Kind;
 
+import java.util.ArrayList;
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/SimpleClassNameCheck.java]: useless change.

File: java-frontend/src/test/files/sym/references/MethodCall.java [hunk 2]
```diff
@@ -341,4 +341,24 @@ public boolean removeAll(Collection<?> collection) {
...
+      .map(line -> line.split("\\s"));
+    stream
+      .forEach(words -> {
+        staticPrint(words[0]);
+        if (foo > words.length || myVar < words.length) {
+          // do something
+        }
+      });
+  }
 }
```
Comment (m-g-sonar) [inline java-frontend/src/test/files/sym/references/MethodCall.java]: missing new line.

File: java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java [hunk 3]
```diff
@@ -906,6 +906,10 @@ public void MethodCall() {
 
     assertThat(result.reference(342, 5)).isSameAs(result.symbol("myMethod", 339));
     assertThat(result.symbol("myMethod", 338).usages()).isEmpty();
+
+    Type streamType = ((VariableTree) result.symbol("stream").declaration()).initializer().symbolType();
+    System.out.println(streamType.name());
```
Comment (m-g-sonar) [inline java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java]: You probably don't want to commit that.

File: java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java [hunk 4]
```diff
@@ -883,7 +883,6 @@ private void registerType(Tree tree, JavaType type) {
 
   private void associateReference(IdentifierTree tree, JavaSymbol symbol) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java]: MAJOR Make "associateReference" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java [hunk 5]
```diff
@@ -881,9 +881,8 @@ private void registerType(Tree tree, JavaType type) {
     types.put(tree, type);
   }
 
-  private void associateReference(IdentifierTree tree, JavaSymbol symbol) {
+  private static void associateReference(IdentifierTree tree, JavaSymbol symbol) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java]: MAJOR Make "associateReference" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**